Applications/SpreadSheet/SpreadSheet.cpp

```diff
@@ -0,0 +1,294 @@
+    {
+        if (interpreter.argument_count() != 1)
+            return interpreter.throw_exception<JS::TypeError>("Expected exactly one argument to parse_cell_name()");
+
+        auto name_value = interpreter.argument(0);
+        if (!name_value.is_string())
+            return interpreter.throw_exception<JS::TypeError>("Expected a String argument to parse_cell_name()");
+
+        auto position = Sheet::parse_cell_name(name_value.as_string().string());
+        if (!position.has_value())
+            return JS::js_undefined();
+
+        auto object = JS::Object::create_empty(interpreter.global_object());
+        object->put("column", JS::js_string(interpreter, position.value().column));
+        object->put("row", JS::Value((unsigned)position.value().row));
+
+        return object;
+    }
+
+private:
+    virtual const char* class_name() const override { return "SheetGlobalObject"; }
+
+    Sheet& m_sheet;
+};
+
+Sheet::Sheet(const StringView& name)
+    : m_name(name)
+    , m_interpreter(JS::Interpreter::create<SheetGlobalObject>(*this))
+{
+    for (size_t i = 0; i < 20; ++i)
+        add_row();
+
+    for (size_t i = 0; i < 16; ++i)
+        add_column();
+
+    auto file_or_error = Core::File::open("/res/js/SpreadSheet/runtime.js", Core::IODevice::OpenMode::ReadOnly);
+    if (!file_or_error.is_error()) {
+        auto buffer = file_or_error.value()->read_all();
+        evaluate(buffer);
+    }
+}
+
+Sheet::~Sheet()
+{
+}
+
+size_t Sheet::add_row()
+{
+    return m_rows++;
+}
+
+String Sheet::add_column()
+{
+    if (m_current_column_name_length == 0) {
+        m_current_column_name_length = 1;
+        m_columns.append("A");
+        return "A";
+    }
+
+    if (m_current_column_name_length == 1) {
+        auto last_char = m_columns.last()[0];
+        if (last_char == 'Z') {
+            m_current_column_name_length = 2;
+            m_columns.append("AA");
+            return "AA";
+        }
+
+        last_char++;
+        m_columns.append({ &last_char, 1 });
+        return m_columns.last();
+    }
+
+    TODO();
+}
+
+void Sheet::update()
+{
+    m_visited_cells_in_update.clear();
+    for (auto& it : m_cells) {
+        if (has_been_visited(it.value))
+            continue;
+        m_visited_cells_in_update.set(it.value);
+        if (it.value->dirty) {
+            // Re-evaluate the cell value, if any.
+            it.value->update({});
+        }
+    }
+
+    m_visited_cells_in_update.clear();
+}
+
+void Sheet::update(Cell& cell)
+{
+    if (has_been_visited(&cell))
+        return;
+
+    m_visited_cells_in_update.set(&cell);
+    cell.update({});
+}
+
+JS::Value Sheet::evaluate(const StringView& source, Cell* on_behalf_of)
+{
+    TemporaryChange cell_change { m_current_cell_being_evaluated, on_behalf_of };
+
+    auto parser = JS::Parser(JS::Lexer(source));
+    if (parser.has_errors())
+        return JS::js_undefined();
+
+    auto program = parser.parse_program();
```

Comment:
Unrelated to your patch, but LibJS should have a convenience function for this :)

Reply:
Not everyone would want to get an `undefined` for parse errors though...?

Reply:
That's true!